cupyx/scipy/special/_gammainc.py

```diff
@@ -0,0 +1,1202 @@
+"""
+The source code here is an adaptation with minimal changes from the following
+files in SciPy's bundled Cephes library:
+
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/const.c
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/igam.c
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/igam.h
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/igami.c
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/lanczos.c
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/lanczos.h
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/mconf.h
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/polevl.h
+https://github.com/scipy/scipy/blob/master/scipy/special/cephes/unity.c
+
+Cephes Math Library Release 2.0:  April, 1987
+Copyright 1984, 1987 by Stephen L. Moshier
+Direct inquiries to 30 Frost Street, Cambridge, MA 02140
+"""
+
+from cupy import _core
+
+from cupyx.scipy.special._digamma import polevl_definition
+from cupyx.scipy.special._zeta import zeta_definition
+
+_zeta_c = zeta_definition
+# comment out duplicate MACHEP definition
+_zeta_c = _zeta_c.replace(
+    "__constant__ double MACHEP", "// __constant__ double MACHEP"
+)
+
+
+# TODO: finish
+
+# See the following on error states:
+#    docs.scipy.org/doc/scipy/reference/generated/scipy.special.errstate.html
+#    scipy/special/sf_error.h
+
+# TODO: which value for MACHEP, MAXLOG from const.c to use?
+
+_misc_preamble = """
+
+// include for CUDART_NAN, CUDART_INF
+#include <math_constants.h>
```

Comment:
See https://github.com/cupy/cupy/pull/5687#issuecomment-907355030.
```suggestion
#include <cupy/math_constants.h>
```